NASA-IMPACT/veda-config
# add playwright

## What am I changing and why

This adds a playwright test suite that generates a list of every catalog and story on the `/data-catalog` and `/stories` routes. It ensures that the pages render with the name  and that there are no javascript errors on those pages.  It is dependent on the changes to the `veda-ui located in https://github.com/NASA-IMPACT/veda-ui/pull/835

A follow-up PR can add the github workflow to trigger the test run on each PR once that veda-ui PR is merged.  The veda-config will also need a mapbox token added as a github secret.

## How to test
run the yarn install to pull in the new dependencies. Run the script `yarn test:e2e` to verify that the tests run locally.**  To verify that tests catch errors, modify pages to break them and ensure that the test suite finds the failure. 

** you may be asked to install the playwright browsers on the first run of the test suite.

## ⚠️ Checks

- [x] I have confirmed that [updating the `veda-ui` submodule](https://github.com/NASA-IMPACT/veda-config/blob/main/docs/DEVELOPMENT.md#development) is needed and **only done so** if that's the case.
